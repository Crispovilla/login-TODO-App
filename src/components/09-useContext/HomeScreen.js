import React, { useContext } from 'react'
import { UserContext } from './UserContext';
import { TodoApp } from '../08-useReducer/TodoApp';
export const HomeScreen = () => {

    const { user } = useContext(UserContext);

    console.log( user );

    return (
        <div>
            <h1>HomeScreen</h1>
            <hr />

            <pre>
                { JSON.stringify( user, null, 3 ) }
                
            </pre>
            <hr />
            <TodoApp />

        </div>
    )
}
