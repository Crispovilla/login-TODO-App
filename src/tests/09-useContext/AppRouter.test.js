import { mount } from "enzyme";
import React from "react";
import { AppRouter } from "../../components/09-useContext/AppRouter";
import { demoUsers } from "../fixtures/demoUsers";
import { UserContext } from "../../components/09-useContext/UserContext";

describe("Pruebas en <AppRouter />", () => {

  const wrapper = mount(
    <UserContext.Provider value={{demoUsers}}>
      <AppRouter />
    </UserContext.Provider>
  );

  test("should Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
