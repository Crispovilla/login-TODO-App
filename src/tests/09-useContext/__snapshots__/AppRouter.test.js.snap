// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> should Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
       useContext 
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
           Home 
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
           About 
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
           Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre />
      <hr />
      <div>
        <h1>
          TodoApp ( 
          0
           ) 
        </h1>
        <hr />
        <div
          className="row"
        >
          <div
            className="col-7"
          >
            <ul
              className="list-group list-group-flush"
            />
          </div>
          <div
            className="col-5"
          >
            Array [
              <h4>
                Agregar TODO
              </h4>,
              <hr />,
              <form
                onSubmit={[Function]}
              >
                <input
                  autoComplete="off"
                  className="form-control"
                  name="description"
                  onChange={[Function]}
                  placeholder="Aprender ..."
                  type="text"
                  value=""
                />
                <button
                  className="btn btn-outline-primary mt-1 btn-block"
                  type="submit"
                >
                  Agregar
                </button>
              </form>,
            ]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
